### ISSUE TITLE
Consume is stalling on getting AlreadyClosedException from RabbitMq

### ISSUE THREAD

**Issue by Fildrance** [rabbit-link/rabbit-link#37]
In some cases:
 * high load (and low resources) 
 * unexpected disconnect
RabbitMq can close connection from server side and rabbit-link can in some way fail to get disconnect event fired. In that case RabbitLink will stop consuming (as server closed connection) but no re-open for connection is called.

For reasons unknow problem is very hard to reproduce and only occures on high load production server, where CPU load and network is high and we are using RabbitMq with load balancer. Other important points are usage of version 0.11.0 and NET Framework.

**Comment by Fildrance** [comment]
As good-looking fix i'd propose https://github.com/interfax-group/rabbit-link/pull/6

**Comment by RaveNoX** [comment]
Hello, thank you for catching and successfully reproducing this bug.
Can you please submit PR for this?

**Comment by Fildrance** [comment]
Yes and no is answer to both questions (catching and reproducing is in question here too :) ). I was getting it only in prod env and had no success in reproducing locally - it really requires some specific state of server (might be over-connecting? sending too many messages? its really hard to guess). I found place where exception thrown is not handled in rabbit-link, and added handling in patch version! pushed it to production and... exception got handled, connection still could not be re-established for quite some time without any specific exception info. Seems like server is refusing connection (again, probably reason lies withing multiple hidden connections. But application itself (according to logs) doesnt seem to spam them. 
I can create PR but interfax-group repo is ahead by several PRs (all of them mentioned in issues here), and i cannot create push branches in this repository - no rights. So i can (and code will be most likely useful) but its not really fixing problem for me here, sadly.
I will try to create branch with fix from certain commit, process won't be pretty. But i guess i knew what i will get when i agreed to merge commits in fork.

**Comment by Fildrance** [comment]
Also sorry for https://github.com/rabbit-link/rabbit-link/pull/35/files - i created PR and then worked on other issues for several months so fork-merge is now looking very ugly and problematic. If we can merge it in as is (after fixing your comments of course) that would be very-very cool.
